add cnn_bilstm package, scripts for testing, initial docs

NickleDave opened the pull request:
* `cnn_bilstm` package containing utils for graph, data manipulation
* scripts to generate learning curves for measuring accuracy given training sets of varying duration
* initial docs -- could be made into a github pages at some point instead of being one giant readme?

yardencsGitHub commented:
Merged the PR.

Currently creating the summary on bird 1 with matlab spectrograms and without normalizations.
I’ll send the results when done. 

Did you receive the new dataset (bird 4)? It is larger (x2) and will allow (hopefully) easier curation.




On May 3, 2018, at 11:02 PM, David Nicholson <notifications@github.com> wrote:

cnn_bilstm package containing utils for graph, data manipulation
scripts to generate learning curves for measuring accuracy given training sets of varying duration
initial docs -- could be made into a github pages at some point instead of being one giant readme?
You can view, comment on, or merge this pull request online at:

  https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4 <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4>
Commit Summary

first commit
fixed many bugs
get summary.py working
add test for utils.reshape_data_for_batching
fix main.py
initial add of config parser
config seems to be working for summary
Merge pull request #3 from NickleDave/add-config-parser
main now works with config, also has logging
make summary.py work with config.ini
fix dumb copy-paste bugs in logging
bugfixes, add spect_scaler
move scaler into main loop so it scales training subset
add config params to choose gpu (to run e.g. two at once)
fix spect_scaler
add utils for creating spectrograms + loading evsonganaly files
add spect_params to main.py and fix bugs dealing with train set duraton
change reshape in setup of tf.nn.bidrect from constant, 512, to `num_hidden`
main takes input_vec_size from spectrograms
fix template_config.ini so it works with spect defaults
add options to save in output_dir and to save every checkpoint file
add summary ops and filewriter, mainly to visualize device placement with tensorboard
add if os.path.isdir to fix FileExistsError in main loop
fix bugs in summary.py due to last set of main.py changes
summary.py saves basically everything, puts in subdir
labels_mapping returned in main.py
load_data accepts labels_mapping
redo main.py + utils so training is same as fcn
change string concat to os.path.join in load_data so glob won't fail
summary.py uses assert do double check .cbin files used
summary.py loads labels_mapping from results_dir
pep8 typo fix
temp add pdb for utils test
add script to load spectrograms from .mat files
factor loading training data out of load_mat_spects
make load_mat_spects part of utils, output consistent with make_data
most of the way through clean up that saves train/test data
new version of main.py works
fix typo in mat_utils, add imports to cnn_bilstm/__init__
fix dumb bug in mat_utils.convert_train_keys
mat_utils.make_data takes training_filenames as arg, also debug both funcs
fix variable name in mat_utils.make_data
save filenames in more idiomatic way
move getting number_song_files to outside training loop
main sets labels_mapping to np.unique(labeled_timebins) from matlab data_dict
fix stupid bug with finding iterations that screwed up all previous results
add error if data_dir not valid for make_data_dict
mat_utils.make_data_from_mat reshapes labels to be same shape as output of make_labeled_timebins
remove f strings, have to go back to 3.5 for tf to work w/GPUs
main saves iter_order for each training
remove loading training data and saving .mat files
add output filename parameters to mat_utils
summary.py loads saved training and test data
main.py loads already generated data_dict from full path or from dir
mat_utils.make_data_from_matlab_spects now makes and saves labels_mapping
remove a stray import pdb
summary.py loads previously saved training and test data
add option results_dir_made_by_main_script under [OUTPUT] in template_config.ini
main.py runs again with matlab spects, now correct labels_mapping
remove unneeded config parameter
fix error handling for .cbins without .not.mats and files with weird labels
fix dumb bug where saved Y_val actually contains X_val
summary.py loads spect_params to make train_data
main.py saves Y_train_subset in scaled_data_dict
first stab at separating making data sets from generating learning curve
second stab
make_data works
add spect_ID_vector to data_dict to use for random grab of subset
make_data and learn_curve both seem to be working
summary works with new make_data configs
Merge pull request #4 from NickleDave/separate-make-data
pull changes from remote
add func to utils to convert back to str labels from labeled_timebins vector
add 'rerun' functionality to learn_curve.py
add context manager needed for pickle.load(train_data) in learn_curve.py
finish adding context manager
bugfix to learn_curve.py so it saves results in correct location
move Levenshtein dist and syllable error rate to metrics.py
add levenshtein and syllable error rate to summary
add labels_mapping to call to convert to labels in summary.py
fix outputs of convert_to_labels
fix stupid bug that added +1 to n_syllables
summary saves pred_and_err_dict including Lev dist and syl err rate
add curvefit.py to cnn_bilstm
had args switched in call to syl_err_rate, fixed
fix bug in convert from timebins to labels that removed repeats
change variable names that didn't get changed--lev dist/syl err rate work now
add code to inference graph to programmatically determine number of hidden units
num_hidden is now determined programaticaally by graphs.py, removed from config/learn_curve.py
add smooth_data and bandpass_filtfilt to cnn_bilstm/evfuncs, for segmenting
add returns to spectrogram doc string in cnn_bilstm/doc_utils
remove unneccesary y from feed dict
cnn_bilstm.__init__.py imports everything
get_inds_for_dur can grab train data subsets starting with least frequent classes
fix up docstring for cnn_bilstm.utils.get_inds_for_dur
fix bugs in utils.get_inds_for_dur affecting 'incfreq' option
fix nefarious indentation bug in cnn_bilstm.utils.get_inds_for_dur
add setup.py so cnn_bilstm can be installed
remove stray 'pdb'
found actual bug in get_inds_for_dur, variable in loop that didn't get deleted
make_data_dicts iteratively gets random subsets of data, and fails gracefully
getting scripts working with canary data, many changes
bugfix: code for labels_mapping caused prediction of nonexistent classes
remove stray pdb, add error checking for n_syllables
make_data.py can now use files from BirdsongRecognition dataset
minor bugfixes to get working with koumura
add determine_unique_labels to cnn_bilstm.koumura_utils
fix summary.py to work with koumura song when labels are str not int
moar bugfixes so summary works with labels from koumura
add READMEs
add more to README_config.md
more README_config.md edits
add more ways in which user can specify spectrogram parameters
move range_str to utils, fix so it returns list of str
stopped trying to load option as type boolean from config
change check for train set duration to allow for floating point error
minor pep8 fixes in cnn_bilstm/mat_utils.py
learn_curve.py makes a subdirectory in /logs for each network trained
getting make_data.py to work with .mat spects
add convert_mat_to_spect to mat_utils
make_spects in utils.py should not crash when checking timebin_dur is consistent across files
make_data now calls mat_utils.convert_mat_to_spect
fix template_config_matlab_spectrograms
mat_utils.convert_mat_to_spect now properly outputs .spect files like utils.make_spects
minor docstring fix in cnn_bilstm.utils.make_data_dicts
make_data calls convert_mat_to_spect with labels_mapping so .spect file has record of it
add error handling for matlab data in learn_curve.py
mat_utils.py maps 'lables' from .mat files to consecutive integers
add vec_translate utility function to utils.py
make_data.py script checks for silent_gap_label in config.ini
add silent_gap_label option to 'DATA' section in template_config_matlab_spectrograms.ini
fix inconsistent variable names in cnn_bilstm/utils.get_inds_for_dur
define newly added options to README_config.md
put new option 'all_labels_are_int' in template_config_matlab_spectrograms.ini
make_data checks for config option 'all_labels_are_int'
remove lines from mat_utils.convert_mat_to_spect that converted labelset from int to str
remove stray import pdb from cnn_bilstm/utils
when all_labels_are_int is True, make_data.py converts only keys in labels_mapping, not labelset
remove stale option 'log_transform' from README_config.md, replace with current option 'transform_type'
clean up README_cnnbilstm_package.md
fix template_config_matlab_spectrograms.ini
add README_summary.md
move README_summary.md into README_cnnbilstm_package.md, add info for learn_curves.py to README_cnnbilstm as well
File Changes

D .DS_Store <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-0> (0)
M .gitignore <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-1> (8)
A README_cnnbilstm_package.md <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-2> (247)
A README_config.md <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-3> (204)
A cnn_bilstm/__init__.py <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-4> (8)
A cnn_bilstm/curvefit.py <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-5> (147)
A cnn_bilstm/evfuncs.py <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-6> (246)
A cnn_bilstm/graphs.py <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-7> (187)
A cnn_bilstm/koumura_utils.py <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-8> (197)
A cnn_bilstm/mat_utils.py <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-9> (229)
A cnn_bilstm/metrics.py <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-10> (81)
A cnn_bilstm/spect_utils.py <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-11> (68)
A cnn_bilstm/utils.py <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-12> (960)
A learn_curve.py <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-13> (479)
A make_data.py <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-14> (185)
A setup.py <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-15> (22)
A summary.py <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-16> (404)
A template_config.ini <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-17> (42)
A template_config_matlab_spectrograms.ini <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-18> (38)
A tests.py <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-19> (55)
Patch Links:

https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4.patch <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4.patch>
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4.diff <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4.diff>
—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AQhbKWejsCOrGuUufFNbK03Ave2LU30qks5tu8TUgaJpZM4TyDfa>.

yardencsGitHub commented:
This is the learning curve I’m getting for bird #1 without normalizations:

I’ll investigate the test set error types .. either from the summary files or from running from saved checkpoints.
Are you reshaping and feeding spectrograms differently or can I use my old scripts (in the notebook) to run from a checkpoint? 

On May 4, 2018, at 8:35 AM, Yarden Cohen <yardenc@bu.edu> wrote:

Merged the PR.

Currently creating the summary on bird 1 with matlab spectrograms and without normalizations.
I’ll send the results when done. 

Did you receive the new dataset (bird 4)? It is larger (x2) and will allow (hopefully) easier curation.




On May 3, 2018, at 11:02 PM, David Nicholson <notifications@github.com <mailto:notifications@github.com>> wrote:

cnn_bilstm package containing utils for graph, data manipulation
scripts to generate learning curves for measuring accuracy given training sets of varying duration
initial docs -- could be made into a github pages at some point instead of being one giant readme?
You can view, comment on, or merge this pull request online at:

  https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4 <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4>
Commit Summary

first commit
fixed many bugs
get summary.py working
add test for utils.reshape_data_for_batching
fix main.py
initial add of config parser
config seems to be working for summary
Merge pull request #3 from NickleDave/add-config-parser
main now works with config, also has logging
make summary.py work with config.ini
fix dumb copy-paste bugs in logging
bugfixes, add spect_scaler
move scaler into main loop so it scales training subset
add config params to choose gpu (to run e.g. two at once)
fix spect_scaler
add utils for creating spectrograms + loading evsonganaly files
add spect_params to main.py and fix bugs dealing with train set duraton
change reshape in setup of tf.nn.bidrect from constant, 512, to `num_hidden`
main takes input_vec_size from spectrograms
fix template_config.ini so it works with spect defaults
add options to save in output_dir and to save every checkpoint file
add summary ops and filewriter, mainly to visualize device placement with tensorboard
add if os.path.isdir to fix FileExistsError in main loop
fix bugs in summary.py due to last set of main.py changes
summary.py saves basically everything, puts in subdir
labels_mapping returned in main.py
load_data accepts labels_mapping
redo main.py + utils so training is same as fcn
change string concat to os.path.join in load_data so glob won't fail
summary.py uses assert do double check .cbin files used
summary.py loads labels_mapping from results_dir
pep8 typo fix
temp add pdb for utils test
add script to load spectrograms from .mat files
factor loading training data out of load_mat_spects
make load_mat_spects part of utils, output consistent with make_data
most of the way through clean up that saves train/test data
new version of main.py works
fix typo in mat_utils, add imports to cnn_bilstm/__init__
fix dumb bug in mat_utils.convert_train_keys
mat_utils.make_data takes training_filenames as arg, also debug both funcs
fix variable name in mat_utils.make_data
save filenames in more idiomatic way
move getting number_song_files to outside training loop
main sets labels_mapping to np.unique(labeled_timebins) from matlab data_dict
fix stupid bug with finding iterations that screwed up all previous results
add error if data_dir not valid for make_data_dict
mat_utils.make_data_from_mat reshapes labels to be same shape as output of make_labeled_timebins
remove f strings, have to go back to 3.5 for tf to work w/GPUs
main saves iter_order for each training
remove loading training data and saving .mat files
add output filename parameters to mat_utils
summary.py loads saved training and test data
main.py loads already generated data_dict from full path or from dir
mat_utils.make_data_from_matlab_spects now makes and saves labels_mapping
remove a stray import pdb
summary.py loads previously saved training and test data
add option results_dir_made_by_main_script under [OUTPUT] in template_config.ini
main.py runs again with matlab spects, now correct labels_mapping
remove unneeded config parameter
fix error handling for .cbins without .not.mats and files with weird labels
fix dumb bug where saved Y_val actually contains X_val
summary.py loads spect_params to make train_data
main.py saves Y_train_subset in scaled_data_dict
first stab at separating making data sets from generating learning curve
second stab
make_data works
add spect_ID_vector to data_dict to use for random grab of subset
make_data and learn_curve both seem to be working
summary works with new make_data configs
Merge pull request #4 from NickleDave/separate-make-data
pull changes from remote
add func to utils to convert back to str labels from labeled_timebins vector
add 'rerun' functionality to learn_curve.py
add context manager needed for pickle.load(train_data) in learn_curve.py
finish adding context manager
bugfix to learn_curve.py so it saves results in correct location
move Levenshtein dist and syllable error rate to metrics.py
add levenshtein and syllable error rate to summary
add labels_mapping to call to convert to labels in summary.py
fix outputs of convert_to_labels
fix stupid bug that added +1 to n_syllables
summary saves pred_and_err_dict including Lev dist and syl err rate
add curvefit.py to cnn_bilstm
had args switched in call to syl_err_rate, fixed
fix bug in convert from timebins to labels that removed repeats
change variable names that didn't get changed--lev dist/syl err rate work now
add code to inference graph to programmatically determine number of hidden units
num_hidden is now determined programaticaally by graphs.py, removed from config/learn_curve.py
add smooth_data and bandpass_filtfilt to cnn_bilstm/evfuncs, for segmenting
add returns to spectrogram doc string in cnn_bilstm/doc_utils
remove unneccesary y from feed dict
cnn_bilstm.__init__.py imports everything
get_inds_for_dur can grab train data subsets starting with least frequent classes
fix up docstring for cnn_bilstm.utils.get_inds_for_dur
fix bugs in utils.get_inds_for_dur affecting 'incfreq' option
fix nefarious indentation bug in cnn_bilstm.utils.get_inds_for_dur
add setup.py so cnn_bilstm can be installed
remove stray 'pdb'
found actual bug in get_inds_for_dur, variable in loop that didn't get deleted
make_data_dicts iteratively gets random subsets of data, and fails gracefully
getting scripts working with canary data, many changes
bugfix: code for labels_mapping caused prediction of nonexistent classes
remove stray pdb, add error checking for n_syllables
make_data.py can now use files from BirdsongRecognition dataset
minor bugfixes to get working with koumura
add determine_unique_labels to cnn_bilstm.koumura_utils
fix summary.py to work with koumura song when labels are str not int
moar bugfixes so summary works with labels from koumura
add READMEs
add more to README_config.md
more README_config.md edits
add more ways in which user can specify spectrogram parameters
move range_str to utils, fix so it returns list of str
stopped trying to load option as type boolean from config
change check for train set duration to allow for floating point error
minor pep8 fixes in cnn_bilstm/mat_utils.py
learn_curve.py makes a subdirectory in /logs for each network trained
getting make_data.py to work with .mat spects
add convert_mat_to_spect to mat_utils
make_spects in utils.py should not crash when checking timebin_dur is consistent across files
make_data now calls mat_utils.convert_mat_to_spect
fix template_config_matlab_spectrograms
mat_utils.convert_mat_to_spect now properly outputs .spect files like utils.make_spects
minor docstring fix in cnn_bilstm.utils.make_data_dicts
make_data calls convert_mat_to_spect with labels_mapping so .spect file has record of it
add error handling for matlab data in learn_curve.py
mat_utils.py maps 'lables' from .mat files to consecutive integers
add vec_translate utility function to utils.py
make_data.py script checks for silent_gap_label in config.ini
add silent_gap_label option to 'DATA' section in template_config_matlab_spectrograms.ini
fix inconsistent variable names in cnn_bilstm/utils.get_inds_for_dur
define newly added options to README_config.md
put new option 'all_labels_are_int' in template_config_matlab_spectrograms.ini
make_data checks for config option 'all_labels_are_int'
remove lines from mat_utils.convert_mat_to_spect that converted labelset from int to str
remove stray import pdb from cnn_bilstm/utils
when all_labels_are_int is True, make_data.py converts only keys in labels_mapping, not labelset
remove stale option 'log_transform' from README_config.md, replace with current option 'transform_type'
clean up README_cnnbilstm_package.md
fix template_config_matlab_spectrograms.ini
add README_summary.md
move README_summary.md into README_cnnbilstm_package.md, add info for learn_curves.py to README_cnnbilstm as well
File Changes

D .DS_Store <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-0> (0)
M .gitignore <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-1> (8)
A README_cnnbilstm_package.md <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-2> (247)
A README_config.md <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-3> (204)
A cnn_bilstm/__init__.py <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-4> (8)
A cnn_bilstm/curvefit.py <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-5> (147)
A cnn_bilstm/evfuncs.py <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-6> (246)
A cnn_bilstm/graphs.py <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-7> (187)
A cnn_bilstm/koumura_utils.py <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-8> (197)
A cnn_bilstm/mat_utils.py <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-9> (229)
A cnn_bilstm/metrics.py <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-10> (81)
A cnn_bilstm/spect_utils.py <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-11> (68)
A cnn_bilstm/utils.py <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-12> (960)
A learn_curve.py <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-13> (479)
A make_data.py <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-14> (185)
A setup.py <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-15> (22)
A summary.py <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-16> (404)
A template_config.ini <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-17> (42)
A template_config_matlab_spectrograms.ini <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-18> (38)
A tests.py <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-19> (55)
Patch Links:

https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4.patch <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4.patch>
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4.diff <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4.diff>
—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AQhbKWejsCOrGuUufFNbK03Ave2LU30qks5tu8TUgaJpZM4TyDfa>.

NickleDave commented:
Yay!  Thanks for accepting the PR.

Did you attach the curve? I don't see it.

Yes, the data is reshaped the same way since that's what the network
requires.
There's a utility function to do it for you,
cnn_bilstm.utils.reshape_data_for_batching

I'm dealing with administrivia this morning but will respond to your edits
later

On May 4, 2018 10:51, "yardencsGitHub" <notifications@github.com> wrote:

This is the learning curve I’m getting for bird #1 without normalizations:

I’ll investigate the test set error types .. either from the summary files
or from running from saved checkpoints.
Are you reshaping and feeding spectrograms differently or can I use my old
scripts (in the notebook) to run from a checkpoint?

On May 4, 2018, at 8:35 AM, Yarden Cohen <yardenc@bu.edu> wrote:

Merged the PR.

Currently creating the summary on bird 1 with matlab spectrograms and
without normalizations.
I’ll send the results when done.

Did you receive the new dataset (bird 4)? It is larger (x2) and will allow
(hopefully) easier curation.




On May 3, 2018, at 11:02 PM, David Nicholson <notifications@github.com
<mailto:notifications@github.com>> wrote:

cnn_bilstm package containing utils for graph, data manipulation
scripts to generate learning curves for measuring accuracy given training
sets of varying duration
initial docs -- could be made into a github pages at some point instead of
being one giant readme?
You can view, comment on, or merge this pull request online at:

https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4
<
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4
>
Commit Summary

first commit
fixed many bugs
get summary.py working
add test for utils.reshape_data_for_batching
fix main.py
initial add of config parser
config seems to be working for summary
Merge pull request #3 from NickleDave/add-config-parser
main now works with config, also has logging
make summary.py work with config.ini
fix dumb copy-paste bugs in logging
bugfixes, add spect_scaler
move scaler into main loop so it scales training subset
add config params to choose gpu (to run e.g. two at once)
fix spect_scaler
add utils for creating spectrograms + loading evsonganaly files
add spect_params to main.py and fix bugs dealing with train set duraton
change reshape in setup of tf.nn.bidrect from constant, 512, to `num_hidden`
main takes input_vec_size from spectrograms
fix template_config.ini so it works with spect defaults
add options to save in output_dir and to save every checkpoint file
add summary ops and filewriter, mainly to visualize device placement with
tensorboard
add if os.path.isdir to fix FileExistsError in main loop
fix bugs in summary.py due to last set of main.py changes
summary.py saves basically everything, puts in subdir
labels_mapping returned in main.py
load_data accepts labels_mapping
redo main.py + utils so training is same as fcn
change string concat to os.path.join in load_data so glob won't fail
summary.py uses assert do double check .cbin files used
summary.py loads labels_mapping from results_dir
pep8 typo fix
temp add pdb for utils test
add script to load spectrograms from .mat files
factor loading training data out of load_mat_spects
make load_mat_spects part of utils, output consistent with make_data
most of the way through clean up that saves train/test data
new version of main.py works
fix typo in mat_utils, add imports to cnn_bilstm/__init__
fix dumb bug in mat_utils.convert_train_keys
mat_utils.make_data takes training_filenames as arg, also debug both funcs
fix variable name in mat_utils.make_data
save filenames in more idiomatic way
move getting number_song_files to outside training loop
main sets labels_mapping to np.unique(labeled_timebins) from matlab
data_dict
fix stupid bug with finding iterations that screwed up all previous results
add error if data_dir not valid for make_data_dict
mat_utils.make_data_from_mat reshapes labels to be same shape as output of
make_labeled_timebins
remove f strings, have to go back to 3.5 for tf to work w/GPUs
main saves iter_order for each training
remove loading training data and saving .mat files
add output filename parameters to mat_utils
summary.py loads saved training and test data
main.py loads already generated data_dict from full path or from dir
mat_utils.make_data_from_matlab_spects now makes and saves labels_mapping
remove a stray import pdb
summary.py loads previously saved training and test data
add option results_dir_made_by_main_script under [OUTPUT] in
template_config.ini
main.py runs again with matlab spects, now correct labels_mapping
remove unneeded config parameter
fix error handling for .cbins without .not.mats and files with weird labels
fix dumb bug where saved Y_val actually contains X_val
summary.py loads spect_params to make train_data
main.py saves Y_train_subset in scaled_data_dict
first stab at separating making data sets from generating learning curve
second stab
make_data works
add spect_ID_vector to data_dict to use for random grab of subset
make_data and learn_curve both seem to be working
summary works with new make_data configs
Merge pull request #4 from NickleDave/separate-make-data
pull changes from remote
add func to utils to convert back to str labels from labeled_timebins vector
add 'rerun' functionality to learn_curve.py
add context manager needed for pickle.load(train_data) in learn_curve.py
finish adding context manager
bugfix to learn_curve.py so it saves results in correct location
move Levenshtein dist and syllable error rate to metrics.py
add levenshtein and syllable error rate to summary
add labels_mapping to call to convert to labels in summary.py
fix outputs of convert_to_labels
fix stupid bug that added +1 to n_syllables
summary saves pred_and_err_dict including Lev dist and syl err rate
add curvefit.py to cnn_bilstm
had args switched in call to syl_err_rate, fixed
fix bug in convert from timebins to labels that removed repeats
change variable names that didn't get changed--lev dist/syl err rate work
now
add code to inference graph to programmatically determine number of hidden
units
num_hidden is now determined programaticaally by graphs.py, removed from
config/learn_curve.py
add smooth_data and bandpass_filtfilt to cnn_bilstm/evfuncs, for segmenting
add returns to spectrogram doc string in cnn_bilstm/doc_utils
remove unneccesary y from feed dict
cnn_bilstm.__init__.py imports everything
get_inds_for_dur can grab train data subsets starting with least frequent
classes
fix up docstring for cnn_bilstm.utils.get_inds_for_dur
fix bugs in utils.get_inds_for_dur affecting 'incfreq' option
fix nefarious indentation bug in cnn_bilstm.utils.get_inds_for_dur
add setup.py so cnn_bilstm can be installed
remove stray 'pdb'
found actual bug in get_inds_for_dur, variable in loop that didn't get
deleted
make_data_dicts iteratively gets random subsets of data, and fails
gracefully
getting scripts working with canary data, many changes
bugfix: code for labels_mapping caused prediction of nonexistent classes
remove stray pdb, add error checking for n_syllables
make_data.py can now use files from BirdsongRecognition dataset
minor bugfixes to get working with koumura
add determine_unique_labels to cnn_bilstm.koumura_utils
fix summary.py to work with koumura song when labels are str not int
moar bugfixes so summary works with labels from koumura
add READMEs
add more to README_config.md
more README_config.md edits
add more ways in which user can specify spectrogram parameters
move range_str to utils, fix so it returns list of str
stopped trying to load option as type boolean from config
change check for train set duration to allow for floating point error
minor pep8 fixes in cnn_bilstm/mat_utils.py
learn_curve.py makes a subdirectory in /logs for each network trained
getting make_data.py to work with .mat spects
add convert_mat_to_spect to mat_utils
make_spects in utils.py should not crash when checking timebin_dur is
consistent across files
make_data now calls mat_utils.convert_mat_to_spect
fix template_config_matlab_spectrograms
mat_utils.convert_mat_to_spect now properly outputs .spect files like
utils.make_spects
minor docstring fix in cnn_bilstm.utils.make_data_dicts
make_data calls convert_mat_to_spect with labels_mapping so .spect file has
record of it
add error handling for matlab data in learn_curve.py
mat_utils.py maps 'lables' from .mat files to consecutive integers
add vec_translate utility function to utils.py
make_data.py script checks for silent_gap_label in config.ini
add silent_gap_label option to 'DATA' section in
template_config_matlab_spectrograms.ini
fix inconsistent variable names in cnn_bilstm/utils.get_inds_for_dur
define newly added options to README_config.md
put new option 'all_labels_are_int' in
template_config_matlab_spectrograms.ini
make_data checks for config option 'all_labels_are_int'
remove lines from mat_utils.convert_mat_to_spect that converted labelset
from int to str
remove stray import pdb from cnn_bilstm/utils
when all_labels_are_int is True, make_data.py converts only keys in
labels_mapping, not labelset
remove stale option 'log_transform' from README_config.md, replace with
current option 'transform_type'
clean up README_cnnbilstm_package.md
fix template_config_matlab_spectrograms.ini
add README_summary.md
move README_summary.md into README_cnnbilstm_package.md, add info for
learn_curves.py to README_cnnbilstm as well
File Changes

D .DS_Store <
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-0>
(0)
M .gitignore <
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-1>
(8)
A README_cnnbilstm_package.md <
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-2>
(247)
A README_config.md <
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-3>
(204)
A cnn_bilstm/__init__.py <
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-4>
(8)
A cnn_bilstm/curvefit.py <
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-5>
(147)
A cnn_bilstm/evfuncs.py <
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-6>
(246)
A cnn_bilstm/graphs.py <
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-7>
(187)
A cnn_bilstm/koumura_utils.py <
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-8>
(197)
A cnn_bilstm/mat_utils.py <
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-9>
(229)
A cnn_bilstm/metrics.py <
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-10>
(81)
A cnn_bilstm/spect_utils.py <
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-11>
(68)
A cnn_bilstm/utils.py <
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-12>
(960)
A learn_curve.py <
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-13>
(479)
A make_data.py <
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-14>
(185)
A setup.py <
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-15>
(22)
A summary.py <
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-16>
(404)
A template_config.ini <
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-17>
(42)
A template_config_matlab_spectrograms.ini <
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-18>
(38)
A tests.py <
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-19>
(55)
Patch Links:

https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4.patch
<
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4.patch
>
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4.diff
<
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4.diff
>
—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub <
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4>,
or mute the thread <
https://github.com/notifications/unsubscribe-auth/AQhbKWejsCOrGuUufFNbK03Ave2LU30qks5tu8TUgaJpZM4TyDfa
>.



—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4#issuecomment-386626044>,
or mute the thread
<https://github.com/notifications/unsubscribe-auth/ALYZig7nD-yrAso-qmPYdF7yvWqGo6VRks5tvGrvgaJpZM4TyDfa>
.

yardencsGitHub commented:
On May 4, 2018, at 11:12 AM, David Nicholson <notifications@github.com> wrote:

Yay! Thanks for accepting the PR.

Did you attach the curve? I don't see it.

Yes, the data is reshaped the same way since that's what the network
requires.
There's a utility function to do it for you,
cnn_bilstm.utils.reshape_data_for_batching

I'm dealing with administrivia this morning but will respond to your edits
later

On May 4, 2018 10:51, "yardencsGitHub" <notifications@github.com <mailto:notifications@github.com>> wrote:

This is the learning curve I’m getting for bird #1 without normalizations:

I’ll investigate the test set error types .. either from the summary files
or from running from saved checkpoints.
Are you reshaping and feeding spectrograms differently or can I use my old
scripts (in the notebook) to run from a checkpoint?

On May 4, 2018, at 8:35 AM, Yarden Cohen <yardenc@bu.edu <mailto:yardenc@bu.edu>> wrote:

Merged the PR.

Currently creating the summary on bird 1 with matlab spectrograms and
without normalizations.
I’ll send the results when done.

Did you receive the new dataset (bird 4)? It is larger (x2) and will allow
(hopefully) easier curation.




On May 3, 2018, at 11:02 PM, David Nicholson <notifications@github.com <mailto:notifications@github.com>
<mailto:notifications@github.com <mailto:notifications@github.com>>> wrote:

cnn_bilstm package containing utils for graph, data manipulation
scripts to generate learning curves for measuring accuracy given training
sets of varying duration
initial docs -- could be made into a github pages at some point instead of
being one giant readme?
You can view, comment on, or merge this pull request online at:

https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4 <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4>
<
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4 <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4>
>
Commit Summary

first commit
fixed many bugs
get summary.py working
add test for utils.reshape_data_for_batching
fix main.py
initial add of config parser
config seems to be working for summary
Merge pull request #3 from NickleDave/add-config-parser
main now works with config, also has logging
make summary.py work with config.ini
fix dumb copy-paste bugs in logging
bugfixes, add spect_scaler
move scaler into main loop so it scales training subset
add config params to choose gpu (to run e.g. two at once)
fix spect_scaler
add utils for creating spectrograms + loading evsonganaly files
add spect_params to main.py and fix bugs dealing with train set duraton
change reshape in setup of tf.nn.bidrect from constant, 512, to `num_hidden`
main takes input_vec_size from spectrograms
fix template_config.ini so it works with spect defaults
add options to save in output_dir and to save every checkpoint file
add summary ops and filewriter, mainly to visualize device placement with
tensorboard
add if os.path.isdir to fix FileExistsError in main loop
fix bugs in summary.py due to last set of main.py changes
summary.py saves basically everything, puts in subdir
labels_mapping returned in main.py
load_data accepts labels_mapping
redo main.py + utils so training is same as fcn
change string concat to os.path.join in load_data so glob won't fail
summary.py uses assert do double check .cbin files used
summary.py loads labels_mapping from results_dir
pep8 typo fix
temp add pdb for utils test
add script to load spectrograms from .mat files
factor loading training data out of load_mat_spects
make load_mat_spects part of utils, output consistent with make_data
most of the way through clean up that saves train/test data
new version of main.py works
fix typo in mat_utils, add imports to cnn_bilstm/__init__
fix dumb bug in mat_utils.convert_train_keys
mat_utils.make_data takes training_filenames as arg, also debug both funcs
fix variable name in mat_utils.make_data
save filenames in more idiomatic way
move getting number_song_files to outside training loop
main sets labels_mapping to np.unique(labeled_timebins) from matlab
data_dict
fix stupid bug with finding iterations that screwed up all previous results
add error if data_dir not valid for make_data_dict
mat_utils.make_data_from_mat reshapes labels to be same shape as output of
make_labeled_timebins
remove f strings, have to go back to 3.5 for tf to work w/GPUs
main saves iter_order for each training
remove loading training data and saving .mat files
add output filename parameters to mat_utils
summary.py loads saved training and test data
main.py loads already generated data_dict from full path or from dir
mat_utils.make_data_from_matlab_spects now makes and saves labels_mapping
remove a stray import pdb
summary.py loads previously saved training and test data
add option results_dir_made_by_main_script under [OUTPUT] in
template_config.ini
main.py runs again with matlab spects, now correct labels_mapping
remove unneeded config parameter
fix error handling for .cbins without .not.mats and files with weird labels
fix dumb bug where saved Y_val actually contains X_val
summary.py loads spect_params to make train_data
main.py saves Y_train_subset in scaled_data_dict
first stab at separating making data sets from generating learning curve
second stab
make_data works
add spect_ID_vector to data_dict to use for random grab of subset
make_data and learn_curve both seem to be working
summary works with new make_data configs
Merge pull request #4 from NickleDave/separate-make-data
pull changes from remote
add func to utils to convert back to str labels from labeled_timebins vector
add 'rerun' functionality to learn_curve.py
add context manager needed for pickle.load(train_data) in learn_curve.py
finish adding context manager
bugfix to learn_curve.py so it saves results in correct location
move Levenshtein dist and syllable error rate to metrics.py
add levenshtein and syllable error rate to summary
add labels_mapping to call to convert to labels in summary.py
fix outputs of convert_to_labels
fix stupid bug that added +1 to n_syllables
summary saves pred_and_err_dict including Lev dist and syl err rate
add curvefit.py to cnn_bilstm
had args switched in call to syl_err_rate, fixed
fix bug in convert from timebins to labels that removed repeats
change variable names that didn't get changed--lev dist/syl err rate work
now
add code to inference graph to programmatically determine number of hidden
units
num_hidden is now determined programaticaally by graphs.py, removed from
config/learn_curve.py
add smooth_data and bandpass_filtfilt to cnn_bilstm/evfuncs, for segmenting
add returns to spectrogram doc string in cnn_bilstm/doc_utils
remove unneccesary y from feed dict
cnn_bilstm.__init__.py imports everything
get_inds_for_dur can grab train data subsets starting with least frequent
classes
fix up docstring for cnn_bilstm.utils.get_inds_for_dur
fix bugs in utils.get_inds_for_dur affecting 'incfreq' option
fix nefarious indentation bug in cnn_bilstm.utils.get_inds_for_dur
add setup.py so cnn_bilstm can be installed
remove stray 'pdb'
found actual bug in get_inds_for_dur, variable in loop that didn't get
deleted
make_data_dicts iteratively gets random subsets of data, and fails
gracefully
getting scripts working with canary data, many changes
bugfix: code for labels_mapping caused prediction of nonexistent classes
remove stray pdb, add error checking for n_syllables
make_data.py can now use files from BirdsongRecognition dataset
minor bugfixes to get working with koumura
add determine_unique_labels to cnn_bilstm.koumura_utils
fix summary.py to work with koumura song when labels are str not int
moar bugfixes so summary works with labels from koumura
add READMEs
add more to README_config.md
more README_config.md edits
add more ways in which user can specify spectrogram parameters
move range_str to utils, fix so it returns list of str
stopped trying to load option as type boolean from config
change check for train set duration to allow for floating point error
minor pep8 fixes in cnn_bilstm/mat_utils.py
learn_curve.py makes a subdirectory in /logs for each network trained
getting make_data.py to work with .mat spects
add convert_mat_to_spect to mat_utils
make_spects in utils.py should not crash when checking timebin_dur is
consistent across files
make_data now calls mat_utils.convert_mat_to_spect
fix template_config_matlab_spectrograms
mat_utils.convert_mat_to_spect now properly outputs .spect files like
utils.make_spects
minor docstring fix in cnn_bilstm.utils.make_data_dicts
make_data calls convert_mat_to_spect with labels_mapping so .spect file has
record of it
add error handling for matlab data in learn_curve.py
mat_utils.py maps 'lables' from .mat files to consecutive integers
add vec_translate utility function to utils.py
make_data.py script checks for silent_gap_label in config.ini
add silent_gap_label option to 'DATA' section in
template_config_matlab_spectrograms.ini
fix inconsistent variable names in cnn_bilstm/utils.get_inds_for_dur
define newly added options to README_config.md
put new option 'all_labels_are_int' in
template_config_matlab_spectrograms.ini
make_data checks for config option 'all_labels_are_int'
remove lines from mat_utils.convert_mat_to_spect that converted labelset
from int to str
remove stray import pdb from cnn_bilstm/utils
when all_labels_are_int is True, make_data.py converts only keys in
labels_mapping, not labelset
remove stale option 'log_transform' from README_config.md, replace with
current option 'transform_type'
clean up README_cnnbilstm_package.md
fix template_config_matlab_spectrograms.ini
add README_summary.md
move README_summary.md into README_cnnbilstm_package.md, add info for
learn_curves.py to README_cnnbilstm as well
File Changes

D .DS_Store <
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-0 <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-0>>
(0)
M .gitignore <
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-1 <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-1>>
(8)
A README_cnnbilstm_package.md <
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-2 <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-2>>
(247)
A README_config.md <
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-3 <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-3>>
(204)
A cnn_bilstm/__init__.py <
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-4 <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-4>>
(8)
A cnn_bilstm/curvefit.py <
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-5 <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-5>>
(147)
A cnn_bilstm/evfuncs.py <
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-6 <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-6>>
(246)
A cnn_bilstm/graphs.py <
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-7 <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-7>>
(187)
A cnn_bilstm/koumura_utils.py <
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-8 <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-8>>
(197)
A cnn_bilstm/mat_utils.py <
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-9 <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-9>>
(229)
A cnn_bilstm/metrics.py <
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-10 <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-10>>
(81)
A cnn_bilstm/spect_utils.py <
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-11 <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-11>>
(68)
A cnn_bilstm/utils.py <
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-12 <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-12>>
(960)
A learn_curve.py <
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-13 <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-13>>
(479)
A make_data.py <
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-14 <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-14>>
(185)
A setup.py <
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-15 <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-15>>
(22)
A summary.py <
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-16 <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-16>>
(404)
A template_config.ini <
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-17 <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-17>>
(42)
A template_config_matlab_spectrograms.ini <
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-18 <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-18>>
(38)
A tests.py <
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-19 <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4/files#diff-19>>
(55)
Patch Links:

https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4.patch <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4.patch>
<
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4.patch <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4.patch>
>
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4.diff <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4.diff>
<
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4.diff <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4.diff>
>
—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub <
https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4 <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4>>,
or mute the thread <
https://github.com/notifications/unsubscribe-auth/AQhbKWejsCOrGuUufFNbK03Ave2LU30qks5tu8TUgaJpZM4TyDfa <https://github.com/notifications/unsubscribe-auth/AQhbKWejsCOrGuUufFNbK03Ave2LU30qks5tu8TUgaJpZM4TyDfa>
>.



—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4#issuecomment-386626044 <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4#issuecomment-386626044>>,
or mute the thread
<https://github.com/notifications/unsubscribe-auth/ALYZig7nD-yrAso-qmPYdF7yvWqGo6VRks5tvGrvgaJpZM4TyDfa <https://github.com/notifications/unsubscribe-auth/ALYZig7nD-yrAso-qmPYdF7yvWqGo6VRks5tvGrvgaJpZM4TyDfa>>
.
—
You are receiving this because you modified the open/close state.
Reply to this email directly, view it on GitHub <https://github.com/yardencsGitHub/tf_syllable_segmentation_annotation/pull/4#issuecomment-386632460>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AQhbKX0dYQDM_mFGcTDO2XcKN_c57VDMks5tvG_rgaJpZM4TyDfa>.
